src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -2710,6 +2710,190 @@ public static void SoapEncodedSerializationTest_Enum_With_SoapEnumOverrides()
+        XmlSerializer ser = new XmlSerializer(myTypeMapping);
+        MyCircularLink value = new MyCircularLink(true);
+        var ms = new MemoryStream();
+        using (var writer = new XmlTextWriter(ms, Encoding.UTF8))
+        {
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartElement("wrapper");
+            ser.Serialize(writer, value);
+            writer.WriteEndElement();
+            ms.Position = 0;
+            string actualOutput = new StreamReader(ms).ReadToEnd();
+            string baseline = "<wrapper>\r\n  <MyCircularLink xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" id=\"id1\">\r\n    <Link href=\"#id2\" />\r\n    <IntValue xsi:type=\"xsd:int\">0</IntValue>\r\n  </MyCircularLink>\r\n  <MyCircularLink id=\"id2\" d2p1:type=\"MyCircularLink\" xmlns:d2p1=\"http://www.w3.org/2001/XMLSchema-instance\">\r\n    <Link href=\"#id3\" />\r\n    <IntValue xmlns:q1=\"http://www.w3.org/2001/XMLSchema\" d2p1:type=\"q1:int\">1</IntValue>\r\n  </MyCircularLink>\r\n  <MyCircularLink id=\"id3\" d2p1:type=\"MyCircularLink\" xmlns:d2p1=\"http://www.w3.org/2001/XMLSchema-instance\">\r\n    <Link href=\"#id1\" />\r\n    <IntValue xmlns:q2=\"http://www.w3.org/2001/XMLSchema\" d2p1:type=\"q2:int\">2</IntValue>\r\n  </MyCircularLink>";
+            Assert.Equal(actualOutput, baseline);
+            writer.Flush();
+            ms.Position = 0;
+            using (var reader = new XmlTextReader(ms))
+            {
+                reader.ReadStartElement("wrapper");
+                var deserialized = (MyCircularLink)ser.Deserialize(reader);
+                Assert.NotNull(deserialized);
+                Assert.Equal(value.Link.IntValue, deserialized.Link.IntValue);
+                Assert.Equal(value.Link.Link.IntValue, deserialized.Link.Link.IntValue);
+                Assert.Equal(value.Link.Link.Link.IntValue, deserialized.Link.Link.Link.IntValue);
+            }
+        }
+    }
+
+    [Fact]
+    [ActiveIssue(15525)]
+    public static void SoapEncodedSerializationTest_Array()
+    {
+        XmlTypeMapping myTypeMapping = new SoapReflectionImporter().ImportTypeMapping(typeof(MyGroup));
+        XmlSerializer ser = new XmlSerializer(myTypeMapping);
+        Thing[] things = new Thing[] { new Thing() { ThingName = "AAA" }, new Thing() { ThingName = "BBB" } };
+        var value = new MyGroup()
+        {
+            GroupName = "MyName",
+            Things = things
+        };
+
+        var ms = new MemoryStream();
+        using (var writer = new XmlTextWriter(ms, Encoding.UTF8))
+        {
+            writer.Formatting = Formatting.Indented;
+            writer.WriteStartElement("wrapper");
+            ser.Serialize(writer, value);
+            writer.WriteEndElement();
+            ms.Position = 0;
+            string actualOutput = new StreamReader(ms).ReadToEnd();
```

Comment:
Same. You should call writer.Flush(); before reading the actualOutput. You would need to update the baseline below as well, which currently does not contain close element.